sarpy/io/phase_history/cphd.py

```diff
@@ -757,3 +794,149 @@ def read_pvp_vector(self, variable, index, the_range=None):
         field_offset, fld_siz = result
         return self._read_pvp_vector(
             pvp_block_offset, pvp_offset, vector_size, field_offset, frm, fld_siz, row_count, the_range)
+
+class CPHDWriter1_0(AbstractWriter):
+    """
+    The CPHD version 1.0 writer.
+    """
+
+    __slots__ = ('_file_name', '_cphd_meta')
+
+    def __init__(self, file_name, cphd_meta):
+        """
+
+        Parameters
+        ----------
+        file_name : str
+        cphd_meta : sarpy.io.phase_history.cphd1_elements.CPHD.CPHDType
+        """
+        self._cphd_meta = cphd_meta
+        super(CPHDWriter1_0, self).__init__(file_name)
+
+    @property
+    def cphd_meta(self):
+        """
+        sarpy.io.phase_history.cphd1_elements.CPHD.CPHDType: The cphd metadata
+        """
+
+        return self._cphd_meta
+
+    def write_file(self, pvp_block, signal_block, support_block=None):
```

Comment:
Is there a case to be made to separate these into separate function calls, and arrange the writer as a context manager? Just so the whole kit and caboodle doesn't have to be simultaneously loaded into ram? Unless it just always is anyways?

Reply:
Good point; I could certainly see that as a way to reduce the memory footprint especially for collects with many channels.  We could then wrap those with this `write_file` for applications with smaller data sizes. Is that something you'd like us to take a crack at?

Reply:
You can, or I can. That's really your preference.

Reply:
Perhaps best if you do to try to maximize consistency with `SICDWriter` and `NITFWriter`? Now that you're mentioning it I see those are probably structured that way as is currenty done in the  complex converter?

Reply:
I suggest using the write_chip for the signal data. You "have to" implement __call__ somehow, anyways. I can take a shot at abstracting it on Monday, then I'll ask for your opinion?